client/layout/universal-navbar-footer/index.tsx

```diff
@@ -1,10 +1,20 @@
 import './style.scss';
+import { getUrlParts } from '@automattic/calypso-url';
+import { getPathParts, getLanguage } from '@automattic/i18n-utils';
 import { useTranslate } from 'i18n-calypso';
 import SocialLogo from 'calypso/components/social-logo';
 import { navigate } from 'calypso/lib/navigate';
 
+function getRealPathName( path: string ) {
+	const urlParts = getUrlParts( path );
+	const locale = getPathParts( urlParts.pathname ).shift();
```

Comment:
This looks fragile because it relies on the fact that the locale is always in the beginning. I'm pretty sure there are areas in the code that have the locale in a different place, and if eventually, we start using that for them, it might cause unexpected bugs for non-en locales.

Reply:
I believe we should cover that. See how `getLocaleFromPathname()` expects it to be at the end, for example:

https://github.com/Automattic/wp-calypso/blob/e60d34fdb303d9db79f91e92c16d860963a1350b/client/boot/locale.js#L13

and for `localizePath` as well:

https://github.com/Automattic/wp-calypso/blob/606db07dc5ca1fded04d8a251b5b93e1d221ab55/client/lib/i18n-utils/localize-path.js#L14

It's a shame we haven't been consistent with where we place the locale, but it's been like that for a while and we should make sure to support either way.